homeassistant/components/tuya/sensor.py

```diff
@@ -107,6 +108,7 @@ class TuyaSensorEntityDescription(SensorEntityDescription):
             key=DPCode.PM25_VALUE,
             name="Particulate matter 2.5 µm",
             device_class=SensorDeviceClass.PM25,
+            native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
```

Comment:
Units are not defined here, they are provided by Tuya.

Reply:
Please provide some documentation for that statement, because that's not the observed behaviour.

Tuya does not provide the units for the air purifier I have.  Not all measurements have units, but some (e.g. line 679) do.  As I understand it, this is the part of the code where the sensor is defined during addition to Home Assistant.  From my local testing, if the measurement unit is not set, this results in the device class being empty (see line 1077).

If you're trying to tell me that my air purifier is buggy and isn't providing information that it should, please tell me how I should either work around or fix that.

Reply:
From your device:

```json
          "pm25": {
            "type": "Integer",
            "value": {
              "unit": "",
              "min": 0,
              "max": 999,
              "scale": 0,
              "step": 1
            }
          },
```

☝️ That was in the diagnostics dump and what is provided by the Tuya API. 
As you can see, Tuya/your device specifically communicates the unit is `""`, which means, not having a unit in Home Assistant is exactly what was communicated.

> Where in the code should I see the unit being propagated from the device?

https://github.com/home-assistant/core/blob/4a62c0c6d15edc37d5b78bc618a3a702442edfd0/homeassistant/components/tuya/sensor.py#L1108

Reply:
Thanks!  Is there a way to provide that value in configuration or something?  I understand the desire to keep the code clean so I'd like to know how to make things work for devices that aren't well-behaved.

Reply:
> Thanks! Is there a way to provide that value in configuration or something? I

No, this is an issue with your Tuya device. Please contact Tuya.